lib/internal/modules/esm/load.js

```diff
@@ -147,6 +149,12 @@ async function defaultLoad(url, context = kEmptyObject) {
     format = 'commonjs-sync';
   }
 
+  if (getOptionValue('--experimental-strip-types') &&
+    StringPrototypeIncludes(format, 'typescript') &&
```

Comment:
A bit more explicit, probably slightly faster.

```suggestion
    (format === 'module-typescript || format === 'commonjs-typescript') &&
```